### ISSUE TITLE
Format whitespace?????

### ISSUE THREAD

**Issue by rublev** [schoero/eslint-plugin-better-tailwindcss#228]
Been wrestling with this for hours. I had prettier and it worked fine but didn't have as good class sorting as this plugin. Now the plugin works and I disabled prettier, but now any sort of template and HTML formatting is out the window and I cannot get it back. it's either I get the classnames sorted nicely, OR I have template formatting. Would appreciate some advice.

<img width="291" height="237" alt="Image" src="https://github.com/user-attachments/assets/0371e03b-15bc-4d3e-8ec3-05b3d52b706d" />

**Comment by rublev** [comment]
I give up. This is fucking impossible. 8 hours later.

<img width="752" height="290" alt="Image" src="https://github.com/user-attachments/assets/7f86da99-e931-4585-8eb7-270cd574143f" />

**Comment by schoero** [comment]
Hey, I’m sorry you’re running into frustration.

From your second screenshot, it looks like the plugin itself is loaded correctly. 
The issue might be with your `tailwindConfig` or css `entryPoint` not being picked up. The plugin usually is able to warn about key options like `entryPoint` if they are misconfigured. Those warnings are emitted once at the beginning, so have a look at the first few of the 507 warnings in your output.

It is also important that you [correctly import tailwindcss](https://tailwindcss.com/docs/installation/framework-guides/nuxt) in your `entryPoint`.

Could you share your ESLint config? Without that, I can only make guesses about what’s happening. Also the versions of Tailwind CSS, ESLint, eslint-plugin-better-tailwindcss and Node.js could be important.

**Comment by rublev** [comment]
> Hey, I’m sorry you’re running into frustration.
> 
> From your second screenshot, it looks like the plugin itself is loaded correctly. The issue might be with your `tailwindConfig` or css `entryPoint` not being picked up. The plugin usually is able to warn about key options like `entryPoint` if they are misconfigured. Those warnings are emitted once at the beginning, so have a look at the first few of the 507 warnings in your output.
> 
> It is also important that you [correctly import tailwindcss](https://tailwindcss.com/docs/installation/framework-guides/nuxt) in your `entryPoint`.
> 
> Could you share your ESLint config? Without that, I can only make guesses about what’s happening. Also the versions of Tailwind CSS, ESLint, eslint-plugin-better-tailwindcss and Node.js could be important.

Lol apologies for spazzing. Turns out I'm a bit of a moron. I rebuilt my app with typescript instead of plain js as well + I started from a fresh copy of vscode (cursor) and all is swell. I'm just running into one final issue that I can't seem to figure out. (Actually literally as I was mid-issue I updated my main.css with tailwind config code from my old project and it cleared up the yellow squiggles under tailwind classes like in screenshot below):

<img width="536" height="354" alt="Image" src="https://github.com/user-attachments/assets/183df363-a1d5-48cd-b895-9fc922aa4602" />

# Files

**~/Library/Application Support/Cursor/User/settings.json**
```json
{
    "eslint.format.enable": true,
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
        "source.fixAll.eslint": "explicit"
    },
    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
    "[jsonc]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "[json]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "[typescript]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "[javascript]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
    "[vue]": {
        "editor.defaultFormatter": "dbaeumer.vscode-eslint"
    },
}
```

**/Users/<USER>/dev/workspaces/ts-vue-project.code-workspace**
```json
{
	"folders": [
		{
			"path": "../site/gitlab/portfolio"
		}
	],
	"settings": {}
}
```

**eslint.config.ts**
```typescript
import antfu from '@antfu/eslint-config'
import eslintPluginBetterTailwindcss from 'eslint-plugin-better-tailwindcss'
import nuxt from './.nuxt/eslint.config.mjs'

export default antfu(
  {
    rules: {
      'vue/block-order': [
        'error',
        {
          order: ['script', 'template', 'style'],
        },
      ],
    },
  },
  {
    plugins: {
      'better-tailwindcss': eslintPluginBetterTailwindcss,
    },
    settings: {
      'better-tailwindcss': {
        entryPoint: 'app/assets/css/main.css',
      },
    },
    rules: {
      ...eslintPluginBetterTailwindcss.configs['recommended-warn'].rules,
      'better-tailwindcss/multiline': ['off'],
    },
  },
  nuxt(),
)
```

**nuxt.config.ts**
```typescript
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  modules: [
    '@nuxt/devtools',
    '@nuxt/fonts',
    '@nuxt/image',
    '@nuxt/scripts',
    '@nuxt/ui',
    '@vueuse/nuxt',
    '@nuxt/eslint',
  ],
  future: {
    compatibilityVersion: 4,
  },
  compatibilityDate: '2025-05-20',
  eslint: {
    config: {
      standalone: false,
    },
  },
  devtools: { enabled: true },
  typescript: {
    typeCheck: true,
    strict: true,
  },
  ui: {
    theme: {
      colors: [
        'primary',
        'secondary',
        'tertiary',
        'info',
        'success',
        'warning',
        'error',
      ],
    },
  },
  css: ['~/assets/css/main.css'],
})
```

**package.json**
```
{
  "name": "nuxt-app",
  "type": "module",
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare",
    "clean": "git clean -fdx -e .env"
  },
  "dependencies": {
    "@antfu/eslint-config": "5.3.0",
    "@commitlint/cli": "19.8.1",
    "@commitlint/config-conventional": "19.8.1",
    "@nuxt/devtools": "2.6.3",
    "@nuxt/eslint": "1.9.0",
    "@nuxt/fonts": "0.11.4",
    "@nuxt/image": "1.11.0",
    "@nuxt/scripts": "0.11.13",
    "@nuxt/ui": "3.3.4",
    "@vueuse/core": "13.9.0",
    "@vueuse/nuxt": "13.9.0",
    "eslint": "^9.36.0",
    "eslint-plugin-better-tailwindcss": "3.7.9",
    "lint-staged": "16.1.6",
    "nuxt": "^4.1.2",
    "simple-git-hooks": "2.13.1",
    "stylelint": "16.24.0",
    "stylelint-config-recommended-vue": "1.6.1",
    "stylelint-config-standard": "38.0.0",
    "vue-tsc": "3.0.7"
  },
  "simple-git-hooks": {
    "pre-commit": "pnpm lint-staged",
    "commit-msg": "pnpm commitlint --edit $1"
  },
  "lint-staged": {
    "*.{js,ts,mjs,vue}": [
      "eslint --fix"
    ],
    "*.{css,vue}": "stylelint --fix"
  }
}
```
**app/components/ProjectCard.vue** my test component
```vue
<script setup>
defineProps({
  href: {
    type: String,
    required: true,
  },
  title: {
    type: String,
    required: true,
  },
  initials: {
    type: String,
    required: true,
  },
  iconBgClass: {
    type: String,
    required: true,
  },
  role: {
    type: String,
    required: true,
  },
  period: {
    type: String,
    required: true,
  },
})
</script>

<template>
  <a
    :href="href"
    target="_blank"
    class="
      rounded-lg border-white/15 p-4 ease-out
      sm:h-[180px] sm:min-w-[198px]
      flex min-w-full flex-1 flex-col border transition-all duration-700
      hover:-translate-y-2 hover:border-white hover:z-10
      hover:rotate-[-1.142deg] hover:skew-x-[-3.142deg] hover:cursor-alias
      hover:border hover:bg-transparent
      hover:shadow-[2px_2px_0px_0px_rgb(255,255,255),2px_3px_0px_0px_rgb(255,255,255)]
    "
  >
    <div class="mb-6 gap-4 flex items-center">
      <span class="text-lg font-semibold text-white">{{ title }}</span>
    </div>
    <div class="mt-auto flex flex-col">
      <p class="text-sm text-gray-200">{{ role }}</p>
      <span class="mt-2 font-mono text-xs text-gray-300">{{ period }}</span>
    </div>
  </a>
</template>
```

**app/assets/css/main.css**
```
@import url('https://fonts.googleapis.com/css2?family=Syne:wght@400;500;600;700;800&family=JetBrains+Mono:wght@400;500;600;700;800&display=swap');
@import 'tailwindcss';
@import 'tw-animate-css';

@custom-variant dark (&:is(.dark *));

@theme inline {
  --color-background: var(--background);
  --color-foreground: var(--foreground);
  --color-card: var(--card);
  --color-card-foreground: var(--card-foreground);
  --color-popover: var(--popover);
  --color-popover-foreground: var(--popover-foreground);
  --color-primary: var(--primary);
  --color-primary-foreground: var(--primary-foreground);
  --color-secondary: var(--secondary);
  --color-secondary-foreground: var(--secondary-foreground);
  --color-muted: var(--muted);
  --color-muted-foreground: var(--muted-foreground);
  --color-accent: var(--accent);
  --color-accent-foreground: var(--accent-foreground);
  --color-destructive: var(--destructive);
  --color-destructive-foreground: var(--destructive-foreground);
  --color-border: var(--border);
  --color-input: var(--input);
  --color-ring: var(--ring);
  --color-chart-1: var(--chart-1);
  --color-chart-2: var(--chart-2);
  --color-chart-3: var(--chart-3);
  --color-chart-4: var(--chart-4);
  --color-chart-5: var(--chart-5);
  --radius-sm: calc(var(--radius) - 4px);
  --radius-md: calc(var(--radius) - 2px);
  --radius-lg: var(--radius);
  --radius-xl: calc(var(--radius) + 4px);
  --color-sidebar: var(--sidebar);
  --color-sidebar-foreground: var(--sidebar-foreground);
  --color-sidebar-primary: var(--sidebar-primary);
  --color-sidebar-primary-foreground: var(--sidebar-primary-foreground);
  --color-sidebar-accent: var(--sidebar-accent);
  --color-sidebar-accent-foreground: var(--sidebar-accent-foreground);
  --color-sidebar-border: var(--sidebar-border);
  --color-sidebar-ring: var(--sidebar-ring);
}

:root {
  --card: oklch(1 0 0);
  --card-foreground: oklch(0.145 0 0);
  --popover: oklch(1 0 0);
  --popover-foreground: oklch(0.145 0 0);
  --primary: oklch(0.205 0 0);
  --primary-foreground: oklch(0.985 0 0);
  --secondary: oklch(0.97 0 0);
  --secondary-foreground: oklch(0.205 0 0);
  --muted: oklch(0.97 0 0);
  --muted-foreground: oklch(0.556 0 0);
  --accent: oklch(0.97 0 0);
  --accent-foreground: oklch(0.205 0 0);
  --destructive: oklch(0.577 0.245 27.325);
  --destructive-foreground: oklch(0.577 0.245 27.325);
  --border: oklch(0.922 0 0);
  --input: oklch(0.922 0 0);
  --ring: oklch(0.708 0 0);
  --chart-1: oklch(0.646 0.222 41.116);
  --chart-2: oklch(0.6 0.118 184.704);
  --chart-3: oklch(0.398 0.07 227.392);
  --chart-4: oklch(0.828 0.189 84.429);
  --chart-5: oklch(0.769 0.188 70.08);
  --radius: 0.625rem;
  --sidebar: oklch(0.985 0 0);
  --sidebar-foreground: oklch(0.145 0 0);
  --sidebar-primary: oklch(0.205 0 0);
  --sidebar-primary-foreground: oklch(0.985 0 0);
  --sidebar-accent: oklch(0.97 0 0);
  --sidebar-accent-foreground: oklch(0.205 0 0);
  --sidebar-border: oklch(0.922 0 0);
  --sidebar-ring: oklch(0.708 0 0);
  --background: oklch(1 0 0);
  --foreground: oklch(0.145 0 0);
}

.dark {
  --background: oklch(0.145 0 0);
  --foreground: oklch(0.985 0 0);
  --card: oklch(0.145 0 0);
  --card-foreground: oklch(0.985 0 0);
  --popover: oklch(0.145 0 0);
  --popover-foreground: oklch(0.985 0 0);
  --primary: oklch(0.985 0 0);
  --primary-foreground: oklch(0.205 0 0);
  --secondary: oklch(0.269 0 0);
  --secondary-foreground: oklch(0.985 0 0);
  --muted: oklch(0.269 0 0);
  --muted-foreground: oklch(0.708 0 0);
  --accent: oklch(0.269 0 0);
  --accent-foreground: oklch(0.985 0 0);
  --destructive: oklch(0.396 0.141 25.723);
  --destructive-foreground: oklch(0.637 0.237 25.331);
  --border: oklch(0.269 0 0);
  --input: oklch(0.269 0 0);
  --ring: oklch(0.439 0 0);
  --chart-1: oklch(0.488 0.243 264.376);
  --chart-2: oklch(0.696 0.17 162.48);
  --chart-3: oklch(0.769 0.188 70.08);
  --chart-4: oklch(0.627 0.265 303.9);
  --chart-5: oklch(0.645 0.246 16.439);
  --sidebar: oklch(0.205 0 0);
  --sidebar-foreground: oklch(0.985 0 0);
  --sidebar-primary: oklch(0.488 0.243 264.376);
  --sidebar-primary-foreground: oklch(0.985 0 0);
  --sidebar-accent: oklch(0.269 0 0);
  --sidebar-accent-foreground: oklch(0.985 0 0);
  --sidebar-border: oklch(0.269 0 0);
  --sidebar-ring: oklch(0.439 0 0);
}

@layer base {
  * {
    border-color: var(--color-border);
    outline-color: rgb(from var(--color-ring) r g b / 0.5);
  }
  html,
  body {
    background-color: var(--color-background);
    color: var(--color-foreground);
    min-height: 100vh;
    font-family: 'Syne', ui-sans-serif, system-ui, sans-serif;
  }

  .font-mono {
    font-family: 'JetBrains Mono', ui-monospace, monospace;
  }
}

@layer base {
  * {
    @apply border-border outline-ring/50;
  }
  body {
    @apply bg-background text-foreground;
  }
}
```

**Comment by rublev** [comment]
Awesome plugin btw this makes my life 1000000x easier do you have a donation link or something?

**Comment by rublev** [comment]
Dangit I'm getting it again and I don't know why lmao

<img width="574" height="163" alt="Image" src="https://github.com/user-attachments/assets/7b820214-50fb-4319-b47f-4cb9964f4906" />

**Comment by rublev** [comment]
figured it out. to have entrypoint work, with my components, i had to disable the "url" portion in my tailwind doc:

```css
@import url("tailwindcss");
@import url("tw-animate-css");
```

to
```css
@import "tailwindcss"
@import "tw-animate-css"
```

**Comment by rublev** [comment]
Hmm it seems like it keeps randomly coming back: 

<img width="784" height="179" alt="Image" src="https://github.com/user-attachments/assets/b5b0bf50-8833-44d2-a85b-d45ff661e4de" />
